src/tests/methods.spec.ts

```diff
@@ -0,0 +1,26 @@
+import { MiddlewareRegistry } from "../MiddlewareRegistry";
+import { NextApiRequest } from "next";
+
+describe('MiddlewareRegistry.methods', () => {
+    it('should add the methods specified to the registry when provided and return ' +
+        'undefined otherwise', async () => {
+        const request = { url: '/api/a' } as NextApiRequest;
```

Comment:
So I think what this is missing (or perhaps a second test case showing this would be good) is to show how requests with different verbs are distinguished.

psuedocode:

```
GIVEN an API route that takes GET or POST, with middlewares A and B respectively

WHEN a GET request is executed

EXPECT only middleware A (associated to GET) should run
```